ext/standard/var.c

```diff
@@ -1052,7 +1052,9 @@ static void php_var_serialize_intern(smart_str *buf, zval *struc, php_serialize_
 	}
 
 	if (UNEXPECTED(php_serialize_check_stack_limit())) {
+#ifdef ZEND_CHECK_STACK_LIMIT
 		zend_call_stack_size_error();
+#endif
 		return;
 	}
```

Comment:
I suggest moving the `zend_call_stack_size_error();` call to `php_serialize_check_stack_limit()` as it's already guarded with ZEND_CHECK_STACK_LIMIT:

``` c
static zend_always_inline bool php_serialize_check_stack_limit(void)
{
#ifdef ZEND_CHECK_STACK_LIMIT
    if (UNEXPECTED(zend_call_stack_overflowed(EG(stack_limit))) {
        zend_call_stack_size_error();
    }
#else
    return false;
#endif
}
```